quantum/process_keycode/process_leader.c

```diff
@@ -70,6 +73,11 @@ bool process_leader(uint16_t keycode, keyrecord_t *record) {
 #    endif
                 return false;
             }
+#    ifndef LEADER_NO_TIMEOUT
```

Comment:
This change seems unrelated to the topic of the PR. 
Can you please either remove it from this PR, or else add an explanation of this part of the change to the PR description?

Reply:
From my description:

> Aside from the features covered in the title, this also fixes a bug. See line 76 in process_leader.c. When LEADER_NO_TIMEOUT is defined and line prcoess_leader.c:55 is run, leading will remain true, breaking the leading mode until a reset occurs.

Happy to promote to its own PR, drop the change, or leave as is.

Reply:
I think it can remain here.